lib/socket.js

```diff
@@ -44,6 +44,11 @@ class Socket extends Duplex {
   setTimeout () {}
 
   destroySoon () {}
+  
+  destroy() {
+    this._socket._destroy();
+    this.writable = false;
```

Comment:
Yeah, I was getting the AssertionError, too. I traced it to `assert(!socket.writable)`. I just set `writable` to false to resolve it.